modules/terraform/apply_test.go

```diff
@@ -119,3 +119,36 @@ func TestTgApplyOutput(t *testing.T) {
 	allOutputs := OutputForKeys(t, options, []string{"str", "list", "map"})
 	assert.Equal(t, allOutputs, map[string]interface{}{"str": "str", "list": []interface{}{"a", "b", "c"}, "map": map[string]interface{}{"foo": "bar"}})
 }
+
+func TestIdempotentNochanges(t *testing.T) {
+	t.Parallel()
+
+	testFolder, err := files.CopyTerraformFolderToTemp("../../test/fixtures/terraform-no-error", t.Name())
+	require.NoError(t, err)
+
+	options := &Options{
+		TerraformDir: testFolder,
+		NoColor:      true,
+	}
+
+	ApplyAndIdempotent(t, options)
+}
+
+func TestIdempotentWithchanges(t *testing.T) {
+	t.Parallel()
+
+	testFolder, err := files.CopyTerraformFolderToTemp("../../test/fixtures/terraform-not-idempotent", t.Name())
+	require.NoError(t, err)
+
+	options := &Options{
+		TerraformDir: testFolder,
+		NoColor:      true,
+	}
+
+	out, err := ApplyAndIdempotentE(t, options)
+
+	_ = out
+
+	require.Error(t, err)
```

Comment:
Might want to check it's the error you expect.

Reply:
Thats a good catch, up to this point I had not ran the tests with -v, it was failing with an error from terraform because I hadn't ran init. Which really surprised me because the other test worked without init. Turns our null resource has a provider. So the test was passing but not working as expected.

Fixed with this `require.EqualError(t, err, "terraform configuration not idempotent")`